### DISCUSSION TITLE
v2.11.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#2644] [General]
## Features

Hey everyone, this is a big release with a lot of great new features!

### Visit metrics

Umami now tracks individual visit metrics. A visit is when a visitor returns to your website multiple times. With this new data, `bounce rate` and `average visit time` are now much more accurate. In the future, we will use this data to introduce user journey analytics and much more.

![image](https://github.com/umami-software/umami/assets/477975/bd39fde5-d364-4aaf-98bd-1845ebf30418)

### Partial matching

Umami can now do partial matching for metrics instead of exact match. This is especially useful when you want to aggregate stats for a set of pages, for example `/blog` or `/docs`. We've added the functionality to the filtering feature on the website overview page.

![image](https://github.com/umami-software/umami/assets/477975/bef981dc-1094-4436-a468-c31c8f923e1f)

As well as the Insights report.

![image](https://github.com/umami-software/umami/assets/477975/22285621-433e-4fbd-8b5c-1b843c899e46)

### UTM report

We are introducing a new report, the UTM report. This report breaks down all your UTM parameters by source, medium, campaign, term and content, making it easy to track the performance of your marketing campaigns.

![image](https://github.com/umami-software/umami/assets/477975/8223d835-f842-470c-89ad-5e7bbb1df8b5)

### Funnel report enhancements

We've updated the styling of the Funnel report to make the metrics easier to read.

![image](https://github.com/umami-software/umami/assets/477975/d64be730-bf2b-4475-8d4e-6dd1da13619d)

You can now add events to your funnels in addition to URLs. Being able to mix URLs and events, you can get a better picture of your conversion rates.

![image](https://github.com/umami-software/umami/assets/477975/d22d3e2c-c7a4-4871-a822-6850a8926eb4)

The Funnel report can also now accept wildcard URL entries. Simply add `*` to the URL, like `/docs*`.

![image](https://github.com/umami-software/umami/assets/477975/1b881da5-82cd-47e8-903d-5e9845603876)

### Referrer icons

Referrer metrics now show icons for the referring domain.

![image](https://github.com/umami-software/umami/assets/477975/77551b2d-7f3b-4686-ac44-a45d79c731b8)

### Exclude search parameters

You can now configure the tracker to not record query parameters. Just add `data-exclude-search=true` to your tracker tag.

### Private mode

We add a new environment variable called `PRIVATE_MODE` which will disable all external network requests for those who are extra privacy conscious. However, this means that website favicons will no longer show because that makes a request to duckduckgo.com. Requests for fonts and images used by the app will still work because they are hosted locally.

## Fixes
- Realtime is re-animating on every event #2583
- Menu MY ACCOUNT & TEAM lost when responsive mobile size #2585
- REMOVE_TRAILING_SLASH behaves "unexpected" #2592
- Clicking on team website from dashboard changes team #2607
- Admin user shows role as unknown #2609
- Completely redundant large dependency #2623
- Search is broken #2628
- This issue still exists in v2.10.2. url encode special characters #2631
- Dashboard: undefined is not an object (evaluating 'D.find(t=>t.value===e).label') #2632
- t.toLowerCase(...).replaceAll is not a function #2635 

## Updates
- Language updates: Chinese, Dutch, Lithuanian
- Upgrade Prisma to `5.11.0`
- Upgraded Next to `14.1.4`

A huge thanks to all the contributors on this release! @fabian-hiller @mobeicanyue @reithose @kiprasmel 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.11.0'>v2.11.0</a>.</em>

**Comment by SmartPhoneLover** [comment]
Hi, after updating from previous release:
![98235738495935](https://github.com/umami-software/umami/assets/42898749/71ee4158-7821-4a4e-a1a8-b40305513aab)

**Reply by HDBear** [reply]
yes facing same issue when installing a fresh version of umami

**Reply by camponez** [reply]
Same here.

**Reply by franciscao633** [reply]
Can you provide the error from the failed migration?

**Reply by camponez** [reply]
I have pasted in the issue #2645

**Reply by xingpingcn** [reply]
same error code

**Comment by appotry** [comment]
the report can't find the website that transfer to the team

that's a bug? @mikecao

**Reply by appotry** [reply]
admin have a website blog. them transfer to the team dev.  admin in the team dev. 
admin can't find the website in it's own report

**Reply by mikecao** [reply]
Can you create an issue with more details?

**Comment by uiii** [comment]
Doesn't the individual visit tracking break the visitors privacy? How can you do that without cookies?

**Reply by Matthijz98** [reply]
https://umami.is/docs/faq still says no cookies.

**Reply by uiii** [reply]
Yes I believe so, I'm just curious how is it done to preserve the privacy.

**Reply by mikecao** [reply]
We use the same hashing method we use to anonymize user data. It's just applied at an hourly interval. Does not use cookies.

**Reply by Matthijz98** [reply]
So it only recognizes users that come back within an hour?

**Reply by mikecao** [reply]
Yes, the visit length is limited to one hour.

**Reply by uiii** [reply]
Thanks for the explanation.

**Comment by wheeleran** [comment]
i'm getting also sorts of errors since updating. this, on the /dashbaord page:

![Screenshot (451)](https://github.com/umami-software/umami/assets/22234084/4b731551-adfb-46b1-b110-b20c531a6c54)

which i can get past by clicking OK, refreshing the page, and removing '/dashboard' manaully from the URL.

the /settings/websites page looks like this:

![Screenshot (452)](https://github.com/umami-software/umami/assets/22234084/5740a57d-0048-4838-a565-ca56f5b09bec)

self-hosted docker Umami on subdomain hosted on synology NAS. 
thanks in advance!
